lib/matplotlib/backends/backend_pgf.py

```diff
@@ -23,37 +23,55 @@
     _create_pdf_info_dict, _datetime_to_pdf)
 from matplotlib.path import Path
 from matplotlib.figure import Figure
+from matplotlib.font_manager import FontProperties
 from matplotlib._pylab_helpers import Gcf
 
 _log = logging.getLogger(__name__)
 
 
+_DOCUMENTCLASS = r"\documentclass{article}"
+
+
 # Note: When formatting floating point values, it is important to use the
 # %f/{:f} format rather than %s/{} to avoid triggering scientific notation,
 # which is not recognized by TeX.
 
 def _get_preamble():
     """Prepare a LaTeX preamble based on the rcParams configuration."""
+    font_size_pt = FontProperties(
+        size=mpl.rcParams["font.size"]
+    ).get_size_in_points()
     return "\n".join([
         # Remove Matplotlib's custom command \mathdefault.  (Not using
         # \mathnormal instead since this looks odd with Computer Modern.)
         r"\def\mathdefault#1{#1}",
         # Use displaystyle for all math.
         r"\everymath=\expandafter{\the\everymath\displaystyle}",
+        # Set up font sizes to match font.size setting.
+        # If present, use the KOMA package scrextend to adjust the standard
+        # LaTeX font commands (\tiny, ..., \normalsize, ..., \Huge) accordingly.
+        # Otherwise, only set \normalsize, manually.
+        r"\IfFileExists{scrextend.sty}{",
+        r"  \usepackage[fontsize=%fpt]{scrextend}" % font_size_pt,
+        r"}{",
+        r"  \renewcommand{\normalsize}{\fontsize{%f}{%f}\selectfont}"
+        % (font_size_pt, 1.2 * font_size_pt),
+        r"  \normalsize",
+        r"}",
         # Allow pgf.preamble to override the above definitions.
         mpl.rcParams["pgf.preamble"],
-        r"\ifdefined\pdftexversion\else  % non-pdftex case.",
-        r"  \usepackage{fontspec}",
         *([
+            r"\ifdefined\pdftexversion\else  % non-pdftex case.",
```

Comment:
Is there a reason for the change here and immediately below?

Reply:
That’s 14a59001. I guess it’s only tangentially related to the rest of this PR, but I noticed that `\usepackage{fontspec}` was always included in the preamble, which is not necessarily desirable when doing custom font setup (i. e. `pgf.rcfonts=False`).